x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -1312,4 +1303,17 @@ protected boolean skipResolved() {
             return true;
         }
     }
+    
+    abstract static class LogicalPlanAnalyzeRule extends AnalyzeRule<LogicalPlan> {
```

Comment:
I'd pick a different name - maybe `DefaultAnalyzeRule` or `BaseAnalyzeRule`.

Reply:
I was struggling with the name, indeed. I didn't particularly like LogicalPlanAnalyzeRule, but didn't find any better alternatives.

Reply:
I like `BaseAnalyzeRule`.